src/pharmpy/model/external/nonmem/update.py

```diff
@@ -579,14 +641,6 @@ def update_statements(model: Model, old: Statements, new: Statements, trans):
     if new_odes is not None:
         old_odes = old.ode_system
         if new_odes != old_odes:
-            colnames, drop, _, _ = parse_column_info(model.internals.control_stream)
-            col_dropped = dict(zip(colnames, drop))
-            if 'CMT' in col_dropped.keys() and not col_dropped['CMT']:
```

Comment:
On one hand, I do agree that that makes sense. However, say that the user decides to drop the column and update the compartmental system, which would change the CMT column numbering. By ignoring this update, if the user then decides to un-drop the CMT column, the mapping from old to new CMT numbers might be incorrect.
So I would think it might be best to update it either way, however, if you don't agree I can of course try to implement the mentioned logic!